lib/IRGen/GenDecl.cpp

```diff
@@ -1050,10 +1051,8 @@ void IRGenerator::emitGlobalTopLevel(llvm::StringSet<> *linkerDirectives) {
     CurrentIGMPtr IGM = getGenModule(wt.getProtocol()->getDeclContext());
     ensureRelativeSymbolCollocation(wt);
   }
-  if (linkerDirectives) {
-    for (auto &entry: *linkerDirectives) {
-      createLinkerDirectiveVariable(*PrimaryIGM, entry.getKey());
-    }
+  for (auto &entry: linkerDirectives) {
+    createLinkerDirectiveVariable(*PrimaryIGM, entry.getKey());
```

Comment:
I know this isn't your bug, but I'm shocked we iterate over a `StringSet` here without sorting it. This cannot be correct in general.

Reply:
Also, I'm pretty sure you don't need to take the `StringRef`s here by reference since they're trivially copyable.

Reply:
> I know this isn't your bug, but I'm shocked we iterate over a StringSet here without sorting it. This cannot be correct in general.

Ouch, good spot! Looks like we also do this in `writeLdAddCFileIfNeeded`. @nkcsgexi, should TBDGen return a vector of symbols instead of a set?

Reply:
> Also, I'm pretty sure you don't need to take the StringRefs here by reference since they're trivially copyable.

It's a (`StringRef`, string length) pair, but yeah I guess `&` is a little overkill

Reply:
@CodaFi is deterministics your concern here or do you have other concerns?

Reply:
Yes. This code builds globals into the generated LLVM module.

Reply:
Switched to returning a vector, how does this look?